### PR TITLE
Zombie SFX

### LINKED ISSUE #9969: Suggestion: Add action button for zombies to produce some zombie line
## Description
Add action button for zombies to produce some zombie line. Should help with mandatory "Grr!" spamming.
See also discussion of issue #9929

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR 
bout time i did some soundwork

Closes #9969 
**Screenshots**

**Changelog**

:cl:
- add: Zombies now have proper groaning sounds when they scream.
- add: Zombies will occasionally audibly groan or say a voice line.
- add: Zombies now groan when they are hit.

### CODE REVIEW

**Comment by EmoGarbage404** [comment]
Closes #9969

File: Content.Server/Zombies/ZombieComponent.cs [hunk 1]
```diff
@@ -11,44 +11,53 @@ public sealed class ZombieComponent : Component
...
         /// </summary>
-        [ViewVariables]
+        [ViewVariables(VVAccess.ReadWrite)]
         public float MinZombieInfectionChance = 0.1f;
 
         /// <summary>
         /// The skin color of the zombie
         /// </summary>
-        [ViewVariables, DataField("skinColor")]
+        [DataField("skinColor")]
```
Comment (metalgearsloth) [inline Content.Server/Zombies/ZombieComponent.cs:33]: Why remove VV?

File: Content.Server/Zombies/ZombieSystem.cs [hunk 2]
```diff
@@ -29,6 +35,23 @@ public override void Initialize()
...
+        private void OnDamage(EntityUid uid, ZombieComponent component, DamageChangedEvent args)
+        {
+            if (!args.DamageIncreased)
+                return;
+
+            if (component.LastDamageGroanAccumulator > 0)
+                return;
+
+            if (_robustRandom.Prob(0.50f))
+                _chat.TrySendInGameICMessage(uid, "generic automated groan", InGameICChatType.Speak, false);
```
Comment (metalgearsloth) [inline Content.Server/Zombies/ZombieSystem.cs:50]: Is this message intended?

File: Content.Server/Zombies/ZombieSystem.cs [hunk 3]
```diff
@@ -88,19 +111,43 @@ private void OnMeleeHit(EntityUid uid, ZombieComponent component, MeleeHitEvent
...
                 }
             }
         }
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+
+            foreach (var zombiecomp in EntityQuery<ZombieComponent>())
+            {
```
Comment (metalgearsloth) [inline Content.Server/Zombies/ZombieSystem.cs:134]: You should have some kind of "ActiveZombieComponent" that gets tracked instead as it's just for the groan timer. Right now it's going to iterate every zombie every tick, dead or alive, so will lead to unnecessary iteration.

File: Content.Server/Zombies/ZombieSystem.cs [hunk 4]
```diff
@@ -88,19 +111,43 @@ private void OnMeleeHit(EntityUid uid, ZombieComponent component, MeleeHitEvent
...
+                zombiecomp.Accumulator += frameTime;
+                zombiecomp.LastDamageGroanAccumulator -= frameTime;
+
+                if (zombiecomp.Accumulator < 5) //generic number
```
Comment (metalgearsloth) [inline Content.Server/Zombies/ZombieSystem.cs:138]: Should use a const or datafield for these kinds of magic numbers.

File: Content.Server/Zombies/ZombieSystem.cs [hunk 5]
```diff
@@ -88,19 +111,43 @@ private void OnMeleeHit(EntityUid uid, ZombieComponent component, MeleeHitEvent
...
+
+                if (!_robustRandom.Prob(zombiecomp.GroanChance))
+                    continue;
+                zombiecomp.Accumulator -= 5; //same generic number
+
+                //either do a random accent line or scream
+                if (_robustRandom.Prob(0.5f))
+                    _chat.TrySendInGameICMessage(zombiecomp.Owner, "generic automated groan", InGameICChatType.Speak, false);
+                else
+                    _vocal.TryScream(zombiecomp.Owner);
```
Comment (metalgearsloth) [inline Content.Server/Zombies/ZombieSystem.cs:149]: Should just make this its own method given it's the same code as above.

File: Content.Server/Zombies/ZombieComponent.cs [hunk 6]
```diff
@@ -11,44 +11,53 @@ public sealed class ZombieComponent : Component
...
         public string AttackArc = "claw";
 
         /// <summary>
         /// The role prototype of the zombie antag role
         /// </summary>
         [ViewVariables, DataField("zombieRoldId", customTypeSerializer: typeof(PrototypeIdSerializer<AntagPrototype>))]
         public readonly string ZombieRoleId = "Zombie";
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float GroanChance = 0.1f;
```
Comment (metalgearsloth) [inline Content.Server/Zombies/ZombieComponent.cs:55]: comment i.e. reflecting it's from the groan accumulator OR from damage.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Zombies/ZombieComponent.cs (hunk 1 above)
Comment (mirrorcult) [reply Content.Server/Zombies/ZombieComponent.cs:36]: datafield already implies viewvariables

**Comment by EmoGarbage404** [comment]
i am going to kill robusttoolbox

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**